src/Middleware/SocialAuthMiddleware.php

```diff
@@ -176,11 +183,13 @@ protected function _handleCallbackAction(ServerRequest $request, Response $respo
 
         $user->unsetProperty($config['fields']['password']);
 
+        $request->session()->write($config['sessionKey'], $user);
+
         if (!$config['userEntity']) {
             $user = $user->toArray();
         }
 
-        $request->session()->write($config['sessionKey'], $user);
```

Comment:
Why did you move up this statement? This would cause the session record to always be entity make the `userEntity` config redundant.

Reply:
I explained my reasoning above, but my reasoning does not exactly match the code change that I made lol. I will revert this, but look at what I stated above and see if it makes sense for some type of small change here.

Reply:
I guess what we need to determine is exactly what `userEntity` is trying to control. The value stored in the session or value sent in the event triggering or both.